### PR TITLE
Remove `empty_genesis_snapshot_matches` test

### PR DESCRIPTION
**Pull request by mwtian**
Changes in Sui Move modules can affect the content of "empty" genesis blob. I have seen folks working on Sui Move get baffled by breaking the `empty_genesis_snapshot_matches` test. If my mental model is correct, where we always try to use the same build of Sui to generate genesis blob and start nodes with the blob, then we don't need to verify the static content of genesis blobs are not changed by a PR.

### CODE REVIEW

**Review by huitseeker — CHANGES_REQUESTED** [review]
> we try to use the same build of Sui to generate a genesis blob and start nodes with the blob

We do not systematically re-genesis with every network deployment, and want to go to a place where it is eventually a breaking change to re-genesis.

The goal of the genesis snapshot is to force signaling to network operators that they will need to re-genesis if they merge the PR that modifies the snapshot.

**Comment by mwtian** [comment]
> > we try to use the same build of Sui to generate a genesis blob and start nodes with the blob
> 
> We do not systematically re-genesis with every network deployment, and want to go to a place where it is eventually a breaking change to re-genesis.
> 
> The goal of the genesis snapshot is to force signaling to network operators that they will need to re-genesis if they merge the PR that modifies the snapshot.

@huitseeker thanks for the info. Just to get more context, when Sui starts to produce a different genesis blob, what is the mode of breakage? e.g. does the change break running network, if we upgrade nodes with the new binary? Or is it only when we want to start a new network, with stored blob generated from a previous Sui binary?

Currently, the genesis blob contains Sui Move modules: https://github.com/MystenLabs/sui/blob/91d613b75fc11ff5ee040284f199f243bd69f498/crates/sui-config/src/genesis.rs#L379. I have seen them changing every week, and I expect them to continue to change for awhile. I'm curious about the compatibility implication here. With the currently logic, most likely we will need to regenerate genesis blob for each Sui release.

cc @tharbert Feel free to weigh in as the network operator, or point me to them. Thanks!

**Comment by huitseeker** [comment]
> I have seen them changing every week, and I expect them to continue to change for awhile. I'm curious about the compatibility implication and mode of breakage here

I am at the moment sitting in a genesis exercise with @sblackshear, @bmwill, @tharbert, @joyqvq. The subject came up, as we noticed we didn't have to re-genesis given our suite of imported commits.

TL;DR, in a few weeks, when we re-genesis on deployment dynamically, then the problem will become a lot smaller. For now, the signal that a PR requires a re-genesis is still super helpful. If that breaks weekly, so be it.

**Comment by mwtian** [comment]
Sounds good. Reverted the test change and updated the PR to just add @tharbert to the CODEWOWNERS. 

Also remove two lines that point to non-existent files. Feel free to let me know what to add back and I can update them in this PR. @huitseeker

**Review by huitseeker — COMMENTED** [review]
sui_core/src/crypto.rs should be /crates/sui-types/src/crypto.rs
sui/src/keystore.rs should be /crates/sui-sdk/src/crypto.rs

And I'll take membership alongside tharbert on the snapshot tests!

**Comment by mwtian** [comment]
> sui_core/src/crypto.rs should be /crates/sui-types/src/crypto.rs sui/src/keystore.rs should be /crates/sui-sdk/src/crypto.rs
> 
> And I'll take membership alongside tharbert on the snapshot tests!

Sounds good, updated!

**Review by huitseeker — APPROVED** [review]
Thanks @mwtian !
